coalahtml/_coalahtml/app/views/files.html

```diff
@@ -1,7 +1,11 @@
 <div class="files">
   <div class="dir-contents" ng-if="fileType === 'dir'">
     <h3>
-    Contents of <span class="filename">{{ fileName }}</span>
+      Contents of
+      <span ng-repeat= "stop in range(len(ROOTHOME), len(fileName) )" class="filename">
+        <a href="#/file/{{ _slice(fileName, 0, stop).join('/') }}">
+          /{{ basename(_slice(fileName, 0, stop).join('/')) }}</a>
+      </span>
```

Comment:
You can make one function that will accept `fileName` as an argument and will return pair of link and respected name to be shown. As far as I know, you should not have any logic in your view. Then here you'll only iterate over the output of that function.